nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,337 @@
+        _temp.add(URI);
+        _temp.add(DB_ACCESS_KEY);
+        _temp.add(CONSISTENCY);
+        _temp.add(DATABASE_NAME);
+        _temp.add(CONTAINER_ID);
+        _temp.add(PARTITION_KEY);
+        descriptors = Collections.unmodifiableList(_temp);
+    }
+
+    protected CosmosClient cosmosClient;
+    protected CosmosContainer container;
+    protected AzureCosmosDBConnectionService connectionService = null;
+
+    @OnScheduled
+    public void createClient(ProcessContext context) throws NullPointerException, CosmosClientException {
+        final ComponentLog logger = getLogger();
+
+        if (context.getProperty(CONNECTION_SERVICE).isSet()) {
+            connectionService = context.getProperty(CONNECTION_SERVICE).asControllerService(AzureCosmosDBConnectionService.class);
+            this.cosmosClient = this.connectionService.getCosmosClient();
+        } else {
+            final String uri, accessKey;
+            uri =  context.getProperty(URI).getValue();
+            accessKey = context.getProperty(DB_ACCESS_KEY).getValue();
+            final ConsistencyLevel clevel;
+            final String selectedConsistency = context.getProperty(CONSISTENCY).getValue();
+
+            switch(selectedConsistency) {
+                case CONSISTENCY_STRONG:
+                    clevel =  ConsistencyLevel.STRONG;
+                    break;
+                case CONSISTENCY_CONSISTENT_PREFIX:
+                    clevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                    break;
+                case CONSISTENCY_SESSION:
+                    clevel = ConsistencyLevel.SESSION;
+                    break;
+                case CONSISTENCY_BOUNDED_STALENESS:
+                    clevel = ConsistencyLevel.BOUNDED_STALENESS;
+                    break;
+                case CONSISTENCY_EVENTUAL:
+                    clevel = ConsistencyLevel.EVENTUAL;
+                    break;
+                default:
+                    clevel = ConsistencyLevel.SESSION;
+            }
+            if (cosmosClient != null) {
+                closeClient();
+            }
+            if(logger.isDebugEnabled()) {
+                logger.debug("Creating CosmosClient");
+            }
+
+            createCosmosClient(uri, accessKey, clevel);
+        }
+        getCosmosDocumentContainer(context);
+    }
+
+    protected void createCosmosClient(final String uri, final String accessKey, final ConsistencyLevel clevel) throws NullPointerException {
+        try {
+            ConnectionPolicy connectionPolicy = ConnectionPolicy.getDefaultPolicy();
+            this.cosmosClient = new CosmosClientBuilder()
+                .endpoint(uri)
+                .key(accessKey)
+                .connectionPolicy(connectionPolicy)
+                .consistencyLevel(clevel)
+                .buildClient();
+        } catch (NullPointerException e) {
+            getLogger().error("Failed to build cosmosClient due to {}",  e);
+            throw e;
+        }
+    }
```

Comment:
What's the scenario where you get a `NullPointerException` here? It seems like something we should handle. Also you don't have to add a `throws NullPointerException`.

Reply:
I had try catch from the implementation with previous SDK. Now we don't need to catch NullPointerException. I dropped the try and catch completely.